src/ipa_pc/mod.rs

```diff
@@ -1365,7 +1365,7 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
 
             // LC(p_1(X),p_2(X),...,p_m(X)) = SUM ( lamda^i * z_i(x) * p_i(X) )
             lc_polynomial +=
-                (cur_challenge, &(labeled_polynomial.polynomial().naive_mul(&Polynomial::from_coefficients_slice(&[z_i_value]))));
+                (cur_challenge * z_i_value, labeled_polynomial.polynomial());
```

Comment:
@phoinic : I did this, as it seems to be more elegant to me. :)
(Compiles, and all tests passed.)

Reply:
Yes, really, thank you =)